src/cluster.c

```diff
@@ -1030,6 +1033,34 @@ getNodeByQuery(client *c, struct serverCommand *cmd, robj **argv, int argc, int
     /* Only valid for sharded pubsub as regular pubsub can operate on any node and bypasses this layer. */
     int pubsubshard_included =
         (cmd_flags & CMD_PUBSUB) || (c->cmd->proc == execCommand && (c->mstate->cmd_flags & CMD_PUBSUB));
+    int is_write_command =
+        (cmd_flags & CMD_WRITE) || (c->cmd->proc == execCommand && (c->mstate->cmd_flags & CMD_WRITE));
+    int is_cross_slot_command =
+        (cmd_flags & CMD_CROSS_SLOT) || (c->cmd->proc == execCommand && (c->mstate->cmd_flags & CMD_CROSS_SLOT));
+
+    /* Cluster is globally down but we got keys? We only serve the request
+     * if it is a read command and when allow_reads_when_down is enabled. */
+    if (!isClusterHealthy()) {
```

Comment:
This is a behavior change. Today we still try to redirect to another node, which could know of a steady quorum and serve the requests. Although not a significant issue, we've known of some issues in AWS where a node was not having issues but we didn't immediately replace it. Having this check at the bottom allows us to redirect away from it.

Reply:
Do we have a chance to add an option to allow client to enable or disable this feature?

Reply:
`CLIENT CAPA` could be used maybe. I still think we should leave this call where it was.

Reply:
> `CLIENT CAPA` could be used maybe. I still think we should leave this call where it was.

I think it works as we expect.